### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by TahenyBELGUITH**
In this milestone  I: 
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
- make the Integration specs

Screenshot 2022-11-02 124425

a code reviewer is appreciated  😄 
@TahenyBELGUITH

### CODE REVIEW

File: spec/features/users_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+         Impedit sit sunt quaerat, odit,')
+  end
+
+  it "displays the user's first 3 posts" do
+    expect(page).to have_content('Post 4')
+    expect(page).to have_content('Post 3')
+    expect(page).to have_content('Post 2')
+    expect(page).not_to have_content('Post 1')
+  end
+
```
Comment (Omar-Muhamad) [inline spec/features/users_show_spec.rb:52]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=see%20the%20user%27s%20username.-,I%20can%20see%20the%20number%20of%20posts%20the%20user%20has%20written.,-I%20can%20see%20the), Please add a Tests that asserts you can see the number of posts the user has written.

File: spec/features/posts_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    expect(page).to have_content(@first_post.text)
+  end
+
+  it 'displays how many comments on a post' do
+    expect(page).to have_content('Comments:5')
+  end
+
+  it 'displays how many likes on a post' do
+    expect(page).to have_content('Likes: 0')
+  end
```
Comment (Omar-Muhamad) [inline spec/features/posts_index_spec.rb:39]: - [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=see%20the%20user%27s%20username.-,I%20can%20see%20the%20number%20of%20posts%20the%20user%20has%20written.,-I%20can%20see%20a), Please add a Test that asserts you can see the number of posts the user has written.
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20first%20comments%20on%20a%20post.), Please add a Test that asserts you can see the first comments on a post.
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20a%20section%20for%20pagination%20if%20there%20are%20more%20posts%20than%20fit%20on%20the%20view.), Please add a Test that asserts you can see a section for pagination if there are more posts than fit on the view.
- [ ] As per this [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=When%20I%20click%20on%20a%20post%2C%20it%20redirects%20me%20to%20that%20post%27s%20show%20page), Please add a Test that redirects to the post's show page when you click on a post.

**Review by Omar-Muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @TahenyBELGUITH,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by andersonlebon — APPROVED** [review]
### Hi @TahenyBELGUITH 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 :clap: :clap: :100: 

nodding-yes

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Good PR description :heavy_check_mark: 

## Optional suggestions
- [ ] Kindly Check the comments under the review._
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------

**Pull request merged**